x-pack/plugins/osquery/server/search_strategy/osquery/index.ts

```diff
@@ -39,7 +39,17 @@ export const osquerySearchStrategyProvider = <T extends FactoryQueryTypes>(
         })
       ).pipe(
         mergeMap((exists) => {
-          const dsl = queryFactory.buildDsl({ ...request, componentTemplateExists: exists });
+          const strictRequest = {
+            factoryQueryType: request.factoryQueryType,
+            filterQuery: request.filterQuery,
+            ...('aggregations' in request ? { aggregations: request.aggregations } : {}),
+            ...('pagination' in request ? { pagination: request.pagination } : {}),
+            ...('sort' in request ? { sort: request.sort } : {}),
+            ...('actionId' in request ? { actionId: request.actionId } : {}),
+            ...('agentId' in request ? { agentId: request.agentId } : {}),
+            componentTemplateExists: exists,
```

Comment:
originally the field `componentTemplateExists` wasn't passed to `es.search()`, only to `queryFactory.buildDsl()`, but now it is passed to both functions - is this intended?